vm/src/builtins/object.rs

```diff
@@ -174,6 +174,11 @@ impl PyBaseObject {
     #[pyclassmethod(magic)]
     fn init_subclass(_cls: PyTypeRef) {}
 
+    #[pyclassmethod(magic)]
+    fn class_getitem(cls: PyTypeRef, _args: FuncArgs) -> PyObjectRef {
+        cls.into_object()
+    }
```

Comment:
@coolreader18 Now that this is in the codebase, the next step is to tackle this problem `TypeError: @runtime_checkable can be only applied to protocol classes, got <class 'typing.SupportsAbs'>`. I don't think a class is inheriting correctly.

Reply:
If I remove the decorators I can see that the protocols are inheriting from `typing._ProtocolMeta`, which is the result of `typing.Protocol[typing.T_co]`. In CPython `typing.Protocol[typing.T_co]` returns an instance of `typing._GenericAlias`.

If I make my own class with `__class_getitem__` the method is never called and `type` is always returned. If the class has a different metaclass that's returned instead.

I added some prints to the `get_item` method in `pyobject.rs`, and from that I can see `vm.get_special_method(obj, "__class_getitem__")` returns the method for `type`, rather than for the class itself. Again, if there's a metaclass its the method for the metaclass. Shouldn't this be calling the method on the class?

Reply:
Note on these changes. Why not add them in a separate PR? They will eventually be added, might as well add them before `typing` (since `typing` might need more time to land, right?).

Reply:
From my perspective, it makes more sense to reduce the number of PRs that have to be merged before this one, i.e. the dependencies of this PR. For really large changes it's good to split out into distinct features, but for relatively small ones it's fine to just have them as commits.